tests/transforms/test_optimization/test_reduce_non_clifford.py

```diff
@@ -0,0 +1,295 @@
+        ops = [gate, gate]
+
+        qs = QuantumScript(ops)
+        (new_qs,), _ = qml.transforms.zx.reduce_non_clifford(qs)
+
+        assert new_qs.operations == []
+
+    @pytest.mark.parametrize(
+        "num_gates, expected_ops",
+        (
+            (1, [qml.S(0)]),
+            (2, [qml.Z(0)]),
+            (4, []),
+        ),
+    )
+    def test_S_gate_simplification(self, num_gates, expected_ops):
+        """Test S gate simplification/cancellation."""
+        ops = [qml.S(0)] * num_gates
+
+        qs = QuantumScript(ops)
+        (new_qs,), _ = qml.transforms.zx.reduce_non_clifford(qs)
+
+        assert new_qs.operations == expected_ops
+
+    @pytest.mark.parametrize(
+        "num_gates, expected_ops",
+        (
+            (1, [qml.T(0)]),
+            (2, [qml.S(0)]),
+            (4, [qml.Z(0)]),
+            (8, []),
+        ),
+    )
+    def test_T_gate_simplification(self, num_gates, expected_ops):
+        """Test T gate simplification/cancellation."""
+        ops = [qml.T(0)] * num_gates
+
+        qs = QuantumScript(ops)
+        (new_qs,), _ = qml.transforms.zx.reduce_non_clifford(qs)
+
+        assert new_qs.operations == expected_ops
+
+    @pytest.mark.parametrize(
+        "params",
+        (
+            (1.7, 0.0),
+            (3.1, -0.5),
+            (0.1, 0.9, -2.8),
+        ),
+    )
+    def test_merge_RX_rotations(self, params):
+        """Test that RX rotation gates are correctly merged together."""
+        ops = [qml.RX(angle, wires=0) for angle in params]
+
+        qs = QuantumScript(ops)
+        (new_qs,), _ = qml.transforms.zx.reduce_non_clifford(qs)
+
+        assert len(new_qs.operations) == 3
+
+        rot = new_qs.operations[1]
+        assert new_qs.operations[0] == qml.H(0)
+        assert isinstance(rot, qml.RZ)
+        assert new_qs.operations[2] == qml.H(0)
+
+        new_angle = rot.parameters[0]
+        exp_angle = np.mod(np.sum(params), 2 * np.pi)
+
+        assert np.isclose(new_angle, exp_angle)
+
+    @pytest.mark.parametrize(
+        "params",
+        (
+            (1.7, 0.0),
+            (3.1, -0.5),
+            (0.1, 0.9, -2.8),
+        ),
+    )
+    def test_merge_RY_rotations(self, params):
+        """Test that RY rotation gates are correctly merged together."""
+        ops = [qml.RY(angle, wires=0) for angle in params]
+
+        qs = QuantumScript(ops)
+        (new_qs,), _ = qml.transforms.zx.reduce_non_clifford(qs)
+
+        assert len(new_qs.operations) == 5
+
+        rot = new_qs.operations[2]
+        assert new_qs.operations[0] == qml.S(0)
+        assert new_qs.operations[1] == qml.H(0)
+        assert isinstance(rot, qml.RZ)
+        assert new_qs.operations[3] == qml.H(0)
+        assert new_qs.operations[4] == qml.adjoint(qml.S(0))
+
+        new_angle = rot.parameters[0]
+        exp_angle = np.mod(np.sum(params), 2 * np.pi)
+
+        assert np.isclose(new_angle, 2 * np.pi - exp_angle)
```

Comment:
Why is there a `-` sign in the relationship between `exp_angle` and `new_angle` 🤔 

Similar suggestion as above to use `qml.equal`, if you want.